DirectProgramming/C++SYCL_FPGA/Tutorials/Features/state_machine/README.md

```diff
@@ -0,0 +1,225 @@
+
+2. Configure the build system for the Agilex® 7 device family, which is the default.
+
+   ```
+   mkdir build
+   cd build
+   cmake ..
+   ```
+
+   You can create a debuggable binary by setting `CMAKE_BUILD_TYPE` to `Debug`:
+
+   ```
+   mkdir build
+   cd build
+   cmake .. -DCMAKE_BUILD_TYPE=Debug
+   ```
+
+   If you want to use the `report`, `fpga_sim`, or `fpga` flows, you should switch the `CMAKE_BUILD_TYPE` back to `Release`:
+
+   ```
+   cmake .. -DCMAKE_BUILD_TYPE=Release
+   ```
+
+   > **Note**: You can change the default target by using the command:
+   >  ```
+   >  cmake .. -DFPGA_DEVICE=<FPGA device family or FPGA part number>
+   >  ```
+   >
+   > Alternatively, you can target an explicit FPGA board variant and BSP by using the following command:
+   >  ```
+   >  cmake .. -DFPGA_DEVICE=<board-support-package>:<board-variant>
+   >  ```
+   >
+   > You will only be able to run an executable on the FPGA if you specified a BSP.
+
+3. Compile the design with the generated `Makefile`. The following build targets are provided, matching the recommended development flow:
+
+   | Target          | Expected Time  | Output                                                                       | Description
+   |:---             |:---            |:---                                                                          |:---
+   | `make fpga_emu` | Seconds        | x86-64 binary                                                                | Compiles the FPGA device code to the CPU. Use the Intel® FPGA Emulation Platform for OpenCL™ software to verify your SYCL code’s functional correctness.
+   | `make report`   | Minutes        | RTL + FPGA reports                                                           | Compiles the FPGA device code to RTL and generates an optimization report that describes the structures generated on the FPGA, identifies performance bottlenecks, and estimates resource utilization. This report will include the interfaces defined in your selected Board Support Package. The generated RTL may be exported to Intel® Quartus Prime software.
+   | `make fpga_sim` | Minutes        | RTL + FPGA reports + x86-64 binary                                           | Compiles the FPGA device code to RTL and generates a simulation testbench. Use the Questa*-Intel® FPGA Edition simulator to verify your design.
+   | `make fpga`     | Multiple Hours | Quartus Place & Route (Full accelerator) + FPGA reports + x86-64 host binary | Compiles the FPGA device code to RTL and compiles the generated RTL using Intel® Quartus® Prime. If you specified a BSP with `FPGA_DEVICE`, this will generate an FPGA image that you can run on the corresponding accelerator board.
+
+   The `fpga_emu`, `fpga_sim` and `fpga` targets produce binaries that you can run. The executables will be called `TARGET_NAME.fpga_emu`, `TARGET_NAME.fpga_sim`, and `TARGET_NAME.fpga`, where `TARGET_NAME` is the value you specify in `CMakeLists.txt`.
```

Comment:
```suggestion
Compile the design. (The provided targets match the recommended development flow.)

Compile for emulation (fast compile time, targets emulates an FPGA device).
make fpga_emu
Generate the HTML optimization reports.
make report
Compile for simulation (fast compile time, targets simulator FPGA device).
make fpga_sim
Compile with Quartus place and route (To get accurate area estimate, longer compile time).
make fpga
```
you can use a simpler text for this (see https://github.com/juny-lim/oneAPI-samples-fork/blob/task_seq_hw_reuse/DirectProgramming/C%2B%2BSYCL_FPGA/Tutorials/Features/experimental/hardware_reuse/README.md)